language/testing-infra/diem-transactional-test-harness/src/diem_test_harness.rs

```diff
@@ -0,0 +1,395 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::{bail, format_err, Result};
+use diem_crypto::{
+    ed25519::{Ed25519PrivateKey, Ed25519PublicKey},
+    ValidCryptoMaterialStringExt,
+};
+use diem_state_view::StateView;
+use diem_types::{
+    access_path::AccessPath,
+    account_config::{
+        self, type_tag_for_currency_code, AccountResource, BalanceResource, XUS_NAME,
+    },
+    chain_id::ChainId,
+    transaction::{
+        Module as TransactionModule, RawTransaction, Script as TransactionScript,
+        ScriptFunction as TransactionScriptFunction, SignedTransaction, Transaction,
+        TransactionStatus,
+    },
+    vm_status::KeptVMStatus,
+};
+use diem_vm::DiemVM;
+use language_e2e_tests::data_store::{FakeDataStore, GENESIS_CHANGE_SET_FRESH};
+use move_binary_format::file_format::{CompiledModule, CompiledScript};
+use move_core_types::{
+    account_address::AccountAddress,
+    gas_schedule::{GasAlgebra, GasConstants},
+    identifier::{IdentStr, Identifier},
+    language_storage::{ModuleId, ResourceKey, TypeTag},
+    move_resource::MoveStructType,
+    transaction_argument::{convert_txn_args, TransactionArgument},
+};
+use move_lang::{shared::verify_and_create_named_address_mapping, FullyCompiledProgram};
+use move_transactional_test_runner::{
+    framework::{run_test_impl, CompiledState, MoveTestAdapter},
+    tasks::{EmptyCommand, InitCommand, SyntaxChoice, TaskInput},
+    vm_test_harness::view_resource_in_move_storage,
+};
+use once_cell::sync::Lazy;
+use std::{
+    collections::{BTreeMap, HashMap},
+    path::Path,
+};
+use structopt::StructOpt;
+
+/// The Diem transaction test adapter.
+///
+/// This differs from the SimpleVMTestAdapter in a few ways to ensure that our tests mimics
+/// production settings:
+///   - It uses a Diem StateView as its storage backend
+///   - It executes transactions through DiemVM, instead of MoveVM directly
+struct DiemTestAdapter<'a> {
+    compiled_state: CompiledState<'a>,
+    storage: FakeDataStore,
```

Comment:
No chance of using `InMemoryStorage` here? 
I can always play around with that if needed

Reply:
One thing is for sure: we want a StateView-based storage backend since that is what the Diem adapter expects. Now it isn't entirely impossible to implement StateView for InMemoryStorage, but I'd like think little more about its interaction with access paths...